src/main/java/org/mule/extensions/jms/internal/connection/JmsTransactionalConnection.java

```diff
@@ -90,6 +88,7 @@ private void executeTransactionAction(String action, SessionAction transactional
     } finally {
       jmsSessionManager.changeTransactionStatus(NONE);
       jmsSessionManager.unbindSession();
+      jmsSession.close();
```

Comment:
i mean a nested finally, to make sure the session is closed even if any of the two statements above fail